module/zfs/arc.c

```diff
@@ -7865,6 +7971,29 @@ arc_init(void)
 
 	arc_prune_taskq = taskq_create("arc_prune", zfs_arc_prune_task_threads,
 	    defclsyspri, 100, INT_MAX, TASKQ_PREPOPULATE | TASKQ_DYNAMIC);
+	if (max_ncpus > 1) {
+		extern uint_t zfs_multilist_num_sublists;
+		/* this is how multilist_create() computes number of sublists */
+		uint_t num_sublists = (zfs_multilist_num_sublists > 0 ?
+		    zfs_multilist_num_sublists : MAX(boot_ncpus, 4));
+		uint_t nthreads;
+
+		if (zfs_arc_evict_threads == 1) {
+			nthreads = MIN((highbit64(max_ncpus) - 1) +
+			    max_ncpus / 64, 16);
+		} else {
+			nthreads = max_ncpus / 2;
+		}
+
+		zfs_arc_evict_threads_max = max_ncpus < 4 ? 1 :
+		    MIN(nthreads, num_sublists);
+
+		if (zfs_arc_evict_threads_max > 1) {
+			arc_evict_taskq = taskq_create("arc_evict",
+			    zfs_arc_evict_threads_max,
+			    defclsyspri, 0, INT_MAX, TASKQ_PREPOPULATE);
+		}
+	}
```

Comment:
If you have only one CPU, as I suspect CI bots might have, `zfs_arc_evict_threads_max` remain zero, which makes `nthreads` and `ntasks` in `arc_evict_state()` also zero, which makes eviction to never progress.

Reply:
We probably had division by 0 in DIV_ROUND_UP() when nthreads == 0.
Updated, should be fixed.

Reply:
I'm sorry, but as I can see it you fixed nothing, but broken even more.  Now without `use_evcttq` `ntasks` will remain 0 and we still won't enter the loop over sublists and evict nothing.

Reply:
The current version might work, but for `!use_evcttq` case we should probably set `ntasks` to `num_sublists` to replicate the old logic, since the current would be suboptimal/flawed in few ways.